@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -553,6 +555,76 @@ function placePointLabelAtCurrentAnchor(
     textCanvas: TextCanvas,
     screenCollisions: ScreenCollisions,
     isRejected: boolean,
+    outScreenPosition: THREE.Vector3
```

Comment:
Please check my comments inside this function first. I would say, this function can be completely removed and replace by calling directly placePointLabelAtAnchor.

Reply:
Please read my comments above, I am trying to do it in opposite way, extract as much as possible from `placePointLabelAtAnchor` to made it simple utility function that may ONLY check if placement is possible. I would even like to remove the screenCollisions handling from there.
This way I will be able to provide some optimizations inside. Moreover I have already some changes that would totally reduce the problem of TextCanvas state -> it's references to label.layoutStyle, but we need to make some smaller steps first - as already agreed - make PR as small as possible.